src/python/WMCore/WorkQueue/WorkQueueBackend.py

```diff
@@ -427,23 +428,28 @@ def availableWork(self, thresholds, siteJobCounts, team=None,
         :return: a tuple with the elements accepted and an overview of job counts per site
         """
         excludeWorkflows = excludeWorkflows or []
-        elements = []
+        acceptedElems = []
         # If there are no sites, punt early.
         if not thresholds:
             self.logger.error("No thresholds is set: Please check")
-            return elements, siteJobCounts
+            return acceptedElems, siteJobCounts
 
         self.logger.info("Current siteJobCounts:")
         for site, jobsByPrio in viewitems(siteJobCounts):
             self.logger.info("    %s : %s", site, jobsByPrio)
 
+        # Find out how many elements are in Available status
+        numAvail = self.queueLength()
+        self.logger.info("Current amount of WQEs in Available status: %s", numAvail)
+
         self.logger.info("Getting up to %d available work from %s", numElems, self.queueUrl)
         self.logger.info("  for team name: %s", team)
-        self.logger.info("  with excludeWorkflows: %s", excludeWorkflows)
+        self.logger.info("  with excludeWorkflows count: %s", len(excludeWorkflows))
+        self.logger.debug("  with excludeWorkflows: %s", excludeWorkflows)
         self.logger.info("  for thresholds: %s", thresholds)
 
         # FIXME: magic numbers
-        docsSliceSize = 1000
+        docsSliceSize = 2500
```

Comment:
This development is needed because there is a flaw in how this list is used, not because there is a "magic" number here. 
I have to check the code, passing variables in workqueue is usually not straight forward, due to the many layers of modules that it has to propagate through.